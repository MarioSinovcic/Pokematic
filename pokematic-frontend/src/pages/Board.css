
.board-page {
    align-content: center;
    flex-direction: row;
    display: flex;
    height: 100vh;
    width: 100vw;
}

.menu {
    display: inline-block;
}

.team-card {
    z-index: 5;
    position: fixed;
    left: 0;
    bottom: 0;
}


.tasks-content {
    position: absolute;
    left: 250px;
}

.todo-status {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    justify-items: center;
    align-content: center;
    margin-top: 50px;
    margin-left: 15px;
}