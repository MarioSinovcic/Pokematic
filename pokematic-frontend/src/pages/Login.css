.login-background {
    justify-content: space-evenly;
    background-image:  url('../images/PokedexBG.png');
    background-size: 100%;
    margin-top:100px;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
    z-index: -10 !important;
    position: fixed;
    opacity: 0.25;
}