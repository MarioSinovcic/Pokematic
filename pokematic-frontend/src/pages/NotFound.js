import React from 'react';

function NotFound() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          404: You've navigated to a non-existing page of Pokematic!
        </p>
      </header>
    </div>
  );
}

export default NotFound;
