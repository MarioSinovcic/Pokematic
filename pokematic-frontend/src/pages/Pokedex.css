.Pokedex{
   height: 100%;
   background-color: whitesmoke;
   /* background-image: url("../images/background_pokeball.png");
   background-position: center;
   background-repeat: no-repeat; */
}

.TeamCard {
    z-index: 5;
    position: fixed;
    left: 0;
    bottom: 0;
}

