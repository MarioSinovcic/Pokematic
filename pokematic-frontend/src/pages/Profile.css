
.board-page {
    align-content: center;
    flex-direction: row;
    display: flex;
    height: 100%;
    width: 100vw;
}

.menu {
    display: inline-block;
}

.profile-card {
    z-index: 5;
    position: fixed;

}


.profile-content {
    left: 250px;
}

.todo-status {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    justify-items: center;
    align-content: center;
    margin-top: 50px;
    margin-left: 15px;
}

