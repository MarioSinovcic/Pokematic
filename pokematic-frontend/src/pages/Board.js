import React from 'react';
import GoalSidebar from './board-components/GoalSideBar';
import TeamCard from '../shared-components/TeamCard';
import StatusCard from './board-components/StatusCard';
import ModalButton from '../shared-components/ModalButton';
import Header from '../shared-components/Header';
import AddIcon from '@material-ui/icons/Add';
import './Board.css'

import {STATUSLIST} from '../constants';
import fakeGoalResponse from '../goalResponse.json';

class Board extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      goalsList: [],
      todoList: [],
      inProgressList: [],
      inReviewList: [],
      doneList: [],
    }
  }

  componentDidMount(){
    this.getTeamGoals();
  }

  getTeamGoals(){
    //replace with api call
    var goalResponse = fakeGoalResponse;

    const gatheredTeamGoals= [];
    var gatheredTasksForGoals= [];

    for (var goal = 0; goal < goalResponse.length; goal++) {
      gatheredTeamGoals.push(goalResponse[goal]);
      var taskArray = goalResponse[goal]["tasks"];

      for(var task = 0; task < taskArray.length; task++){
        var currentTask = taskArray[task];
        currentTask["goalName"] = goalResponse[goal]["name"];
        gatheredTasksForGoals.push(currentTask);
      }
    }
    this.sortTasks(gatheredTasksForGoals);
    
    this.setState({
      goalsList: gatheredTeamGoals
    })
  }

  sortTasks(tasks){
    const gatheredTodoList = [];
    const gatheredInProgressList = [];
    const gatheredInReviewList = [];
    const gatheredDoneList = [];
    
    for (var i = 0; i < tasks.length; i++) {
      if(tasks[i]["status"] === STATUSLIST[0]){
        gatheredTodoList.push(tasks[i]);
      }
      if(tasks[i]["status"] === STATUSLIST[1]){
        gatheredInProgressList.push(tasks[i]);
      }
      if(tasks[i]["status"] === STATUSLIST[2]){
        gatheredInReviewList.push(tasks[i]);
      }
      if(tasks[i]["status"] === STATUSLIST[3]){
        gatheredDoneList.push(tasks[i]);
      }
    }
        
    this.setState({
      todoList: gatheredTodoList,
      inProgressList: gatheredInProgressList,
      inReviewList: gatheredInReviewList,
      doneList: gatheredDoneList,
    })
  }

  render(){
    return (
        <div>
          <div className="board-page">
          <Header />
          <div className="team-card">
                <TeamCard />
          </div>
            <div className="menu">
            <GoalSidebar goalsList={this.state.goalsList}/>
            </div>
            <div className="tasks-content">
              <div className="todo-status">
              <StatusCard statusTitle={"TODO"} taskList={this.state.todoList}/>
              <StatusCard statusTitle={"IN PROGRESS"} taskList={this.state.inProgressList}/>
              <StatusCard statusTitle={"IN REVIEW"} taskList={this.state.inReviewList}/>
              <StatusCard statusTitle={"DONE"} taskList={this.state.doneList}/>
              </div>
            </div>
            <div className="new-task-button">
              <ModalButton icon={<AddIcon style={{fontSize: "35px"}}/>} theme="dark" type="new-task"/>
            </div>
          </div>
        </div>
     );
  }
}

export default Board;
