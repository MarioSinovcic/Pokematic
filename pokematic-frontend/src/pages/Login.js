import React from 'react';
import Header from './shared-components/Header';
import { Typography, Button } from '@material-ui/core';
import * as colors from './shared-components/colors'
import { Link } from 'react-router-dom';
import auth0Client from '../Auth0/Auth';
import './Login.css';

class Login extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
    }
  }

  render() {
    return (
      <div className="App">
        <Header currentPage=""/>

        <div>
          <div className="login-background"></div>
          <div className="login-space">
            <div className="login-card-shape">
              <div className="login-base-shape">
                <div className="login-card-title">
                  <Typography className="login-card-title">Log In</Typography>
                </div>
                <div className="credentials-shape">
                  <Link to={"/profile"} replace style={{ textDecoration: 'none' }} >
                    <Button style={{ backgroundColor: colors.fire, height: "50px", width: "225px", fontSize: "60px", marginTop: "-125px", border: "white 3px solid" }}
                      className="team-label pokemon-label"
                      onClick={() => auth0Client.signIn()}>
                      Enter Pokematic!
                 </Button>
                  </Link>
                </div>
                <div className="pokeball-img-login"></div>

              </div>
              <div className="login-edge">
                <div className="login-cut"></div>
                <div className="login-bottom"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}



export default Login;
