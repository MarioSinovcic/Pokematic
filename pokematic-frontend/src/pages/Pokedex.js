import React from 'react';
import PokedexList from './pokedex-components/PokedexList'
import Header from '../shared-components/Header'
import TeamCard from '../shared-components/TeamCard';
import './Pokedex.css';
import { connect } from 'react-redux';
import { changeCollection } from '../actions/actions';
import { Typography, Grid, Switch } from '@material-ui/core';

class Pokedex extends React.Component {

  state = {
    teamName: this.props.match.params.teamName,
    pokemonCollection: [],
  }

  componentWillMount() {
    // TEMPORARY TEAM
    this.props.changeCollection([this.props.pokemonData[54], this.props.pokemonData[103]]);
  } 
  
  switchPokemon(event) {

    if (!event.target.checked) { 
      this.setState({
        pokemonCollection: [this.props.pokemonData[54], this.props.pokemonData[103]],
      })

      this.props.changeCollection([this.props.pokemonData[54], this.props.pokemonData[103]]);


    } else {
      this.setState({
        pokemonCollection: this.props.pokemonData,
      })

      this.props.changeCollection(this.props.pokemonData);
      
    }
  };



  render() {

    const handleSwitch = this.switchPokemon.bind(this)

    return (
      <div className="pokedex">

        <div>
          <div className="header">
            <Header currentPage={"/pokedex"} teamName={this.state.teamName}/>
          </div>
          <div className="filter-shape">
            <div className="left-cut" />
            <div className="filter">
              <Typography component="div">
                <Grid component="label" container alignItems="center" spacing={1}>
                  <Grid item className="switch-text">COLLECTION</Grid>
                  <Grid item>
                    <Switch
                      color=""
                      name="checkedC"
                      onChange={(event) => handleSwitch(event)}
                      checked={this.props.pokemonCollection === this.props.pokemonData} />
                  </Grid>
                  <Grid item className="switch-text">ALL POKEMON</Grid>
                </Grid>
              </Typography>
            </div>
            <div className="right-cut" />
          </div>
        </div>
        <div className="team-card">
          <TeamCard teamName={this.state.teamName}/>
        </div>
        <div>
          {this.props.pokemonCollection[0] ? <PokedexList pokemonCollection={this.props.pokemonCollection} /> : ""}
        </div>
      </div >
    );
  }
}

const mapStateToProps = (state) => {
  return {
    pokemonData: state.pokemonData,
    pokemonCollection: state.pokemonCollection,
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    changeCollection: (collection) => {
      dispatch(changeCollection(collection))
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Pokedex);