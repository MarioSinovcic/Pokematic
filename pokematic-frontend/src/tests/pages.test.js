import React from 'react';
import { shallow } from 'enzyme';
import Board from '../pages/Board';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import rootReducers from '../reducers';
import NotFound from '../pages/NotFound';
import TeamSearch from '../pages/TeamSearch';
import Routes from '../router/Routes';
import Login from '../pages/Login';
import Profile from '../pages/Profile';
import Pokedex from '../pages/Pokedex';

//The following code includes examples for snapshot testing components.
//Please try following the naming convetions shown in these examples

//Docs : https://enzymejs.github.io/enzyme/


describe('pages', () => {

    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })

    describe('snapshot tests', () => {
        it('displays error page', () => {
            const notFound = renderer.create(<Provider store={store}>
                <NotFound />
            </Provider>).toJSON();
            expect(notFound).toMatchSnapshot();
        })

        it('team search', () => {
            const team = renderer.create(<Provider store={store}>
                <TeamSearch />
            </Provider>).toJSON();
            expect(team).toMatchSnapshot();
        })
    })
})

describe('Snapshot pages tests', () => {
    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })



    // TODO: test router pages
    describe('Board', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));

            const wrapper = shallow(<Provider store={store}><Board /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })

    });

    describe('Login', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));


            const wrapper = shallow(<Provider store={store}><Login /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })
    });

    describe('Profile', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));


            const wrapper = shallow(<Provider store={store}><Profile /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })
    });

    describe('Pokedex', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));


            const wrapper = shallow(<Provider store={store}><Pokedex /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })
    });


    describe('Board', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));


            const wrapper = shallow(<Provider store={store}><Board /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })

    });

    describe('Incorrect Link', () => {

        //this tests if the component renders without errors
        it('should render without throwing an error', () => {
            jest.mock('react-router-dom', () => ({
                useHistory: () => ({
                    push: jest.fn(),
                }),
            }));

            const wrapper = shallow(<Provider store={store}><NotFound /></Provider>);


            expect(wrapper).toMatchSnapshot();

        })
    });

});

