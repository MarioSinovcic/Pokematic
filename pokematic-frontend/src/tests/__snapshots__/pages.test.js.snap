// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot pages tests Board should render without throwing an error 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Board) />
</ContextProvider>
`;

exports[`Snapshot pages tests Board should render without throwing an error 2`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Board) />
</ContextProvider>
`;

exports[`Snapshot pages tests Incorrect Link should render without throwing an error 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <NotFound />
</ContextProvider>
`;

exports[`Snapshot pages tests Login should render without throwing an error 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Login />
</ContextProvider>
`;

exports[`Snapshot pages tests Pokedex should render without throwing an error 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Pokedex) />
</ContextProvider>
`;

exports[`Snapshot pages tests Profile should render without throwing an error 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Profile />
</ContextProvider>
`;

exports[`pages snapshot tests displays error page 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      404 Cheif
    </p>
  </header>
</div>
`;

exports[`pages snapshot tests team search 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      TeamSearch
    </p>
  </header>
</div>
`;
